### PR TITLE
Mapping spring legacy props

### PR DESCRIPTION
**Pull request by yiliuTo**
Map legacy spring properties of KeyVault, Cosmos and Storage to the current ones. The mapping relationship is recorded in file sdk/spring/azure-spring-cloud-autoconfigure/src/main/resources/legacy-property-mapping.properties.

Implemented by using an EnvironmentPostProcessor to convert all legacy properties detected from the environment to the associtaed current ones, and then put them into environment.

When Key Vault secret starter is used, the predeing processor will be executed again after KeyVaultEnvironmentPostProcessor to convert legacy properties load from Key Vault if exist.

### CODE REVIEW

**Comment by yiliuTo** [comment]
@stliu please help to review this pr when you are available, thanks

File: sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/utils/PropertyLoader.java [hunk 1]
```diff
@@ -37,4 +41,20 @@ private static String getProperty(String file, String property) {
...
     }
+
+    /**
+     * To load a {@link Properties} Object from the properties file under classpath
+     *
+     * @param file The source file to load properties from.
+     * @throws UncheckedIOException If an I/O error occurs.
+     * @return The {@link Properties} Object.
+     */
+    public static Properties loadProperties(String file) {
```
Comment (stliu) [inline sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/utils/PropertyLoader.java:52]: the implementation doesn't match with method name

**Review by stliu — APPROVED** [review]

File: sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/keyvault/PostLegacyPropertyEnvironmentPostProcessor.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+     * the current, and create a new {@link Properties} to store the converted current properties of each key vault
+     * property source.
+     *
+     * @param legacyToCurrentMap A {@link Properties} contains a map of all legacy properties and associated current properties.
+     * @return A {@link Properties} to store mapped current properties
+     */
+    @Override
+    protected Properties convertLegacyPropertyToCurrent(Properties legacyToCurrentMap) {
+        Properties properties = new Properties();
+        List<KeyVaultPropertySource> propertySourceList = getKeyVaultPropertySourceList();
```
Comment (stliu) [inline sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/keyvault/PostLegacyPropertyEnvironmentPostProcessor.java:51]: how about this:

1. getKeyVaultPropertySourceList returns a Stream ( change the method name as well )
2. Stream.sorted(Collections.reverseOrder())

personally i think this has more readability

File: sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/autoconfigure/unity/PreLegacyPropertyEnvironmentPostProcessor.java [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+        return legacyToCurrentMap;
+    }
+
+    /**
+     * Load all possible Key Vault names from property "spring.cloud.azure.keyvault.order" if existed. Otherwise load
+     * from legacy property "azure.keyvault.order".
+     *
+     * @return A string with all Key Vaults names concatenated by commas, or null if no Key Vault names specified.
+     */
+    private String getMultipleKeyVaultNames() {
```
Comment (stliu) [inline sdk/spring/azure-spring-cloud-autoconfigure/src/main/java/com/azure/spring/autoconfigure/unity/PreLegacyPropertyEnvironmentPostProcessor.java:79]: it should be taking care in this method that return an array of names

**Pull request merged**